### PR TITLE
Reworks Cling Hivemind

### PR DESCRIPTION
**Pull request by Coolrune206**
## What Does This PR Do

Removes the Changeling Hivemind from the Changeling base-kit, along with the DNA upload/download ability, and packages the two together as a 2 DNA point purchase. Additionally, removes the ability to hivemind interrogate someone.
Also unintentionally fixes a minor bug with augmented eyesight.
### Hivemind changes
Changeling was, for a while, in a pretty bad state as an antagonist. Hivemind was rather helpful when Changelings were individually weak and thus encouraging them to work together was quite a good idea. However, now with recent Changeling buffs they are significantly stronger and capable of standing on their own, yet the use of Hivemind to form Changeling murder-balls has only increased (predictably, as it's so damn effective at just wiping all resistance to you off the station). There's been some discussion about how to handle this in https://www.paradisestation.org/forum/topic/23547-changeling-hivemind-discussion-thread/ and in #coding-chat, and this solution was proposed as an alternative to flat out deletion of the Cling Hivemind. If this isn't enough or people don't bother buying it, perhaps we can go ahead with that but for now, it's a start.
This has been set at 2 DNA points because 1 DNA point would be far too cheap. It's a completely secure communication grid with none of the downsides that the traitor comms have, so it's a little bit more comparatively expensive.

### DNA upload/download changes
These have been grouped with the hivemind changes as it makes logical sense, both in-lore and for balance. As per lore, if you don't have access to the hivemind to speak/listen, you probably don't have access to it to upload/download DNA strings. As per balance, if we removed hivemind but didn't alter this Clings could probably just spam their own identities into this at the start of the round and immediately know who the other Changelings were anyways.

### Hivemind interrogation removal
This was done for similar reasons to the DNA upload/download, except this feature is also hardly used and would be difficult to implement alongside needing a separate ability to even be able to speak in the hivemind yourself.

## Images of changes

image
image
image

image
image
image

## Testing

See images above.
## Changelog
:cl:
tweak: Moves Changeling Hivemind and DNA sharing to a 2 DNA cost purchase, and removes hivemind interrogation
/:cl:

### CODE REVIEW

**Comment by Gatchapod** [comment]
Before anyone argues this should be 1 DNA point, consider how much power changelings gain from teaming up.
Two traitors working together can still be relatively easily shot to death using basic laser guns.
Two changelings working together is a serious nightmare that requires a well-cooperating security team to take down, and cremating them is even worse. If they destroy the cremator on top of teaming up, security will likely lose most of their officers trying to handle that. The only survivors will be those who did not show up in the chapel.

**Comment by SteelSlayer** [comment]
Can you also remove the cling datum `is_linking` var, and the mind `linglink` var?

**Comment by SpringSkipper** [comment]
I feel like it should only cost 1 point.
Consider that the equivalent traitor option, the syndie comms, costs the equivalent of 1 evolution point (2 TC out of 20) and also gives **full access to all departmental and command communications**. Further, traitors have more to gain by collaborating with eachother due to things like surpluses and agent IDs.

No one will buy the cling hivemind for 2 points when it's essentially a complete gamble that is utterly useless if no one else gets it (unlike syndie comms which still lets you listen in on command and security even if no one else buys it and is, again, half the price).

You need to add a lot more functionality to it, or else just delete it so that it isn't a noob trap.

**Comment by Contrabang** [comment]
I get removing the comms, but I don't like the removal of the DNA Hivemind, as its quite useful for not getting spotted and being able to store your target's DNA. I don't want to spend 2 points a round just for the hivemind ability, I would prefer if they were split up. At most, if all the changelings did upload their own DNA to identify each other (which they wouldn't all do because some like to stay anonymous), they would still need to meet up in person, similar to traitor codewords. If anything, I think this is just going to make changelings more violent, so they can purchase it, meet up, then kill a bunch of people to absorb so they can get their abilities back, which I think would be a helluva lot more toxic meta.

**Comment by Rurik123** [comment]
As a note with DNA Upload/Absorb removal, this will make it so clings have no way around the DNA cap, with old DNA identities going "stale" after a certain point, and removed upon transform. While mostly this just hinders your starter identity, there will be rare occasions where someone kills target first, then goes for the 8 genomes, only to find out upon reaching 8 their target identity was wiped for being stale.

Ideally that clusterfuck of a mechanic could also be stripped, but that is out of scope for this PR.

> I feel like it should only cost 1 point.
> Consider that the equivalent traitor option, the syndie comms, costs the equivalent of 1 evolution point (2 TC out of 20) and also gives **full access to all departmental and command communications**. Further, traitors have more to gain by collaborating with eachother due to things like surpluses and agent IDs and the ability to exchange telecrystals.

I don't know how you reached the conclusion traitors have more to gain by collaborating with each other. Clings, by mere potential of their current status in balance, have far more to gain from teamwork, since with every number added they become exponentially stronger. 10 traitors in a conga is killable. 4 clings in a conga is not. 

Also keep in mind, traitors cannot rebuy gear after killing someone. Clings can. If hivemind is empty and the cling is the only one who bought it, one absorb and they can rebuy to better suit their solo needs, whereas traitors are stuck with their comms. 

The two point cost is more than justified.

**Comment by SpringSkipper** [comment]
> As a note with DNA Upload/Absorb removal, this will make it so clings have no way around the DNA cap, with old DNA identities going "stale" after a certain point, and removed upon transform. While mostly this just hinders your starter identity, there will be rare occasions where someone kills target first, then goes for the 8 genomes, only to find out upon reaching 8 their target identity was wiped for being stale.
> 
> Ideally that clusterfuck of a mechanic could also be stripped, but that is out of scope for this PR.
You can't wipe your impersonation target for being stale (or shouldn't be able to, at least. Cling code not working as intended wouldn't be a shocker). It goes into a second 'protected DNA' list in the code that doesn't get checked for staleness.

**Comment by hal9000PR** [comment]
I disagree with all the comments on this pr but gatchas.

I stg coolrune do not change this pr cos of the haters

**Comment by AffectedArc07** [comment]
> Also, while we're here, let it be known that the "balance team" is keeping a bug in the game so changelings are even weaker than they should be.

Sorry what?

If you had eyes you would likely see the PR is `Awaiting Review` not `Awaiting Approval`. 
image

Attempting to belittle people with "emphasis quotes" will get you nowhere. Your attitude on the git has been notoriously shit on multiple PRs, and its getting ridiculous. 

I advise you change your attitude before I change your posting rights to the non-existent variant.

**Comment by 1080pCat** [comment]
> The attached screenshot is just...ugh. You know people _will_ accidentally expose themselves thanks to the change and you're...doing nothing to stop it? At least make :g tell you to buy the ability and not send the damn message if you're a changeling. No need to screw people over for something that was perfectly standard before this PR.

I do agree with this. it is already an issue that malf ai can go _;b okay time to start the plasma flood_ and just be, screwed over by something thats super easy to miss. 
90% of changlings powers are already a noob trap, intentionally acknowledging a glaring issue that will likely trip up the most robust clings, and not even taking any steps to mitigate it being an issue is.. frankly.. worrying? 'idedplsnerf'? i dunno how to explain this ending part, but im sure you get my point.

TLDR: you're intentionally creating a noob trap for cling players :gatto:

**Comment by farie82** [comment]
> I don't think tone policing will get us anywhere. Is there anything in my argument that you find issue with?

The general tone of your comment has been very bossy and dickish.

> Also, while we're here, let it be known that the "balance team" is keeping a bug in the game so changelings are even weaker than they should be.

This part is just unneeded. The decision there was made in that PR for the reason to keep a fix/refactor separate from a balance change. We prefer modularity in PRs. The part of it being a balance thing is due to the fact that the current changeling balancing is based on that bug/inconsistency/weirdness.

> The attached screenshot is just...ugh. You know people will accidentally expose themselves thanks to the change and you're...doing nothing to stop it? At least make :g tell you to buy the ability and not send the damn message if you're a changeling. No need to screw people over for something that was perfectly standard before this PR.

You're basically blaming the author here for being negligent. You do make a suggestion, but please learn how to make a constructive suggestion without attacking the author.

Now please learn from this event and take AAs advise

**Comment by matttheficus** [comment]
Unless a way is added to "save" your roundstart and target's DNA is added, this will lead to too many jank things with the Hivemind Channel ability. The communications aspect I think is alright, however, and is something that I'd want to see added anyway. 

But, given the jankiness of stored DNA I cant approve this unless a Fix/"Store DNA" ability is added.

**Comment by Coolrune206** [comment]
> Unless a way is added to "save" your roundstart and target's DNA is added, this will lead to too many jank things with the Hivemind Channel ability. The communications aspect I think is alright, however, and is something that I'd want to see added anyway.
> 
> But, given the jankiness of stored DNA I cant approve this unless a Fix/"Store DNA" ability is added.

As discussed on discord, your target's (for escape with identity) DNA is always protected and cannot go stale. I've gone ahead and added your roundstart DNA to this, even though I think we really should encourage Changelings to go for the target's DNA for its value in being protected.

**Review by Sirryan2002 — APPROVED** [review]

**Review by farie82 — APPROVED** [review]
